File: models/transaction.py
from google.appengine.ext import ndb

from models.decimalProperty import DecimalProperty

class Transaction(ndb.Model):
  userid = ndb.StringProperty(required=True)
  date = ndb.DateProperty(required=True)
  payee = ndb.StringProperty(required=True)
  description = ndb.StringProperty(required=True)
  outflow = DecimalProperty(required=True)
  inflow = DecimalProperty(required=True)

File: web/transaction.dart
class Transaction {
  String id;
  String date;
  String payee;
  String description;
  int inflow; // inflow in cents. outflows are just negative values.
  
  Transaction(this.id, this.date, this.payee, this.description, this.inflow);
  
  String amount() {
    return (this.inflow/100).toStringAsFixed(2);
  }
}